alembic/script/base.py

```diff
@@ -554,13 +554,57 @@ def run_env(self) -> None:
     def env_py_location(self) -> str:
         return str(Path(self.dir, "env.py"))
 
-    def _append_template(self, src: Path, dest: Path, **kw: Any) -> None:
-        with util.status(
-            f"Appending to existing {dest.absolute()}",
-            **self.messaging_opts,
-        ):
+
+    def _append_template(
+        self, template_path: Path, output_path: Union[str, Path], **kw: Any
+    ) -> None:
+        """
+        Append content from a Mako template to an existing file.
+        
+        Special handling for TOML files to ensure proper formatting between
+        sections, addressing issue #1679.
+        
+        Args:
+            template_path: Path to the template file
+            output_path: Path to the output file to append to
+            **kw: Keyword arguments for template rendering
+        """
+        output_path = Path(output_path)
+        
+        # For TOML files, we need special handling to ensure proper spacing
+        if output_path.suffix.lower() == '.toml' and output_path.exists():
+            # Read existing content
+            with open(output_path, 'r', encoding=self.output_encoding) as f:
+                existing_content = f.read()
+            
+            # Check if we need to add newlines before appending
+            existing_content = existing_content.rstrip()
```

Comment:
this is not really a "check", this just truncates what was there so that it can add two newlines unconditionally